lib/sockjs.js

```diff
@@ -26,19 +25,27 @@ global.__shinyserverdebug__ = {
 
 // options.disableProtocols can be an array of protocols to remove from the
 // whitelist
-exports.createFactory = function(options) {
+exports.createFactory = function(protocolChooser, options) {
   return function(url, context, callback) {
     if (!callback) throw new Error("callback is required");
 
     url = pathParams.reorderPathParams(url, ["n", "o", "t", "w", "s"]);
 
-    let whitelist = [];
-    require("./protocol-chooser").whitelist.forEach(prot => {
+    var whitelist = [];
```

Comment:
Oh that's anti climactic, I thought I was about to learn something new about ES6 :)